molecad/data/core/downloader.py

```diff
@@ -0,0 +1,267 @@
+    """
+    Функция, которая форматирует первую часть URL-адреса для запроса в базы данных Pubchem. Эта
+    часть определяет, какие записи следует использовать в качестве темы для запроса. Функция
+    форматирует параметры для дальнейшей передачи в функцию ``build_url``.
+    :param domain: принимает значения из класса ``Domain``.
+    .. note:: В текущей версии сервиса доступен поиск только по базе данных ``Compound``.
+    :param namespace: принимает значения в зависимости от определенного ранее параметра
+    ``domain``. В общем случае может иметь простой и составной характер, поэтому значение должно
+    быть предварительно обработано функцией  ``joined_namespaces``.
+    :param identifiers: передаваемое значение должно быть целым числом или строкой; если необходимо
+    указать последовательность идентификаторов, то они должны быть предварительно переданы в
+    функцию ``joined_identifiers``.
+    :return: строка, отформатированная по типу "<domain>/<namespace>/<identifiers>",
+    которая является первой частью URL-адреса.
+    """
+    return f"{domain}/{namespace}/{identifiers}"
+
+
+def operation_specification(operation: str, tags: str = None) -> str:
+    """
+    Часть URL-адреса указывает службе Pubchem, что делать с данными, определенными в первой
+    части URL-адреса, например, вы можете получить конкретные свойства соединений. Функция
+    форматирует параметры для дальнейшей передачи в функцию ``build_url``.
+    :param operation:  принимает значения в зависимости от определенного ранее параметра
+    ``domain``. Если значение не определено, то по умолчанию извлекается вся запись.
+    :param tags: определяется только в случае если ``operation`` == ``Operation.PROPERTY``,
+    иначе не указывается; строка содержит перечень из запрашиваемых тегов, доступных для данной
+    операции, которые предварительно были переданы в функцию ``join_w_comma``.
+    :return: строка, которая является второй частью URL-адреса.
+    """
+    if tags is None:
+        return f"{operation}"
+    else:
+        return f"{operation}/{tags}"
+
+
+def build_url(input_spec: str, operation_spec: str, output_format: str) -> str:
+    """
+    Команда генерирует строку URL-адреса, необходимую для выполнения запроса в службу PubChem.
+    Структура URL-адреса состоит из трех частей, которые предварительно были приведены к
+    надлежащему формату.
+    :param input_spec: значение полученное из функции ``input_specification``.
+    :param operation_spec: значение полученное из функции ``operation_specification``.
+    :param output_format: формат выходных данных, принимающий значение из класса ``Out``.
+    :return: URL-адрес, используемый в качестве аргумента при вызове функции ``request_data``,
+    которая осуществляет запрос в базу данных Pubchem.
+    """
+    return f"{BASE_URL}/{input_spec}/{operation_spec}/{output_format}"
+
+
+def joined_namespace(prefix: str, suffix: Optional[str] = None) -> str:
```

Comment:
Просто на будущее, если эти функции не предполагаются для публичного использования, состоит их именовать так:

```suggestion
def __joined_namespace(prefix: str, suffix: Optional[str] = None) -> str:
```

Reply:
> Просто на будущее, если эти функции не предполагаются для публичного использования, состоит их именовать так:

все, которые не импортируются? я кажется припоминаю что-то такое... а там именно два подчеркивания, а не одно?

Reply:
Я не настаиваю, просто можно так явно обозначить, что предполагается использовать вне этого модуля, а что нет